crates/bevy_reflect/bevy_reflect_derive/src/field_attributes.rs

```diff
@@ -94,54 +94,53 @@ pub(crate) fn parse_field_attrs(attrs: &[Attribute]) -> Result<ReflectFieldAttr,
-        )),
-        Meta::NameValue(pair) if pair.path.is_ident(DEFAULT_ATTR) => {
-            if let Expr::Lit(ExprLit {lit: Lit::Str(lit_str), ..}) = &pair.value {
-                args.default = DefaultBehavior::Func(lit_str.parse()?);
+fn parse_meta(args: &mut ReflectFieldAttr, meta: ParseNestedMeta) -> Result<(), syn::Error> {
+    if meta.path.is_ident(DEFAULT_ATTR) {
+        // Allow:
+        // - `#[reflect(default)]`
+        // - `#[reflect(default = "path::to::func")]`
+        match args.default {
+            DefaultBehavior::Required => {
+                if meta.input.peek(Token![=]) {
+                    let lit = meta.value()?.parse::<LitStr>()?;
+                    args.default = DefaultBehavior::Func(lit.parse()?);
+                } else {
+                    args.default = DefaultBehavior::Default;
+                }
                 Ok(())
             }
-            else {
-                Err(syn::Error::new(
-                    pair.span(),
-                    format!("expected a string literal containing the name of a function, but found: {}", pair.to_token_stream()),
-                ))?
-            }
-        }
-        Meta::NameValue(pair) => {
-            let path = &pair.path;
-            Err(syn::Error::new(
-                path.span(),
-                format!("unknown attribute parameter: {}", path.to_token_stream()),
-            ))
+            _ => Err(meta.error(format!("only one of [{:?}] is allowed", [DEFAULT_ATTR]))),
         }
-        Meta::List(list) if !list.path.is_ident(REFLECT_ATTRIBUTE_NAME) => {
-            Err(syn::Error::new(list.path.span(), "unexpected property"))
+    } else if meta.path.is_ident(IGNORE_ALL_ATTR) {
+        // Allow:
+        // - `#[reflect(ignore)]`
+        match args.ignore {
+            ReflectIgnoreBehavior::None => {
+                args.ignore = ReflectIgnoreBehavior::IgnoreAlways;
+                Ok(())
+            }
+            _ => Err(meta.error(format!(
+                "only one of [{:?}] is allowed",
+                [IGNORE_ALL_ATTR, IGNORE_SERIALIZATION_ATTR]
+            ))),
         }
-        Meta::List(list) => {
-            if let Ok(meta) = list.parse_args() {
-                parse_meta(args, &meta)?;
+    } else if meta.path.is_ident(IGNORE_SERIALIZATION_ATTR) {
+        // Allow:
+        // - `#[reflect(skip_serializing)]`
+        match args.ignore {
+            ReflectIgnoreBehavior::None => {
+                args.ignore = ReflectIgnoreBehavior::IgnoreSerialization;
+                Ok(())
             }
-            Ok(())
+            _ => Err(meta.error(format!(
+                "only one of [{:?}] is allowed",
+                [IGNORE_ALL_ATTR, IGNORE_SERIALIZATION_ATTR]
+            ))),
         }
+    } else {
+        Err(meta.error(format!(
+            "unknown attribute, expected {:?}",
+            [DEFAULT_ATTR, IGNORE_ALL_ATTR, IGNORE_SERIALIZATION_ATTR]
```

Comment:
Would be probably worthwhile to add the name of the unknown attribute to the error message.

Reply:
It's fine if the span already points out the faulty attribute. No need to add it then.